src/filter_tree/filter_tree.c

```diff
@@ -418,6 +418,155 @@ void FilterTree_DeMorgan(FT_FilterNode **root) {
 	_FilterTree_DeMorgan(root, 0);
 }
 
+// This method returns if an arithmetic expression is valid to be use in filter tree compaction.
+static inline bool _FT_FilterNode_constnat_expression(AR_ExpNode *exp) {
+	// TODO: Handle injected parames.
+	return exp->type == AR_EXP_OPERAND && exp->operand.type == AR_EXP_CONSTANT;
+}
+
+// Compacts an expression node. Return if this node can be used in compression.
+static inline bool _FilterTree_Compact_Exp(FT_FilterNode *node) {
+	return _FT_FilterNode_constnat_expression(node->exp.exp);
+}
+
+// In place set an existing filter tree node to expression node.
+static inline void _FilterTree_In_Place_Set_Exp(FT_FilterNode *node, SIValue v) {
+	node->t = FT_N_EXP;
+	node->exp.exp = AR_EXP_NewConstOperandNode(v);
+}
+
+// Compacts 'AND' condition node.
+static bool _FilterTree_Compact_And(FT_FilterNode *node) {
+	// Try to compact left and right children.
+	bool is_lhs_const = FilterTree_Compact(node->cond.left);
+	bool ihs_rhs_const = FilterTree_Compact(node->cond.right);
+	// If both are not compactable, this node is not compactable.
+	if(!is_lhs_const && !ihs_rhs_const) return false;
+	// Compact to 'false' in case at least one child is 'false'
+	if(is_lhs_const) {
+		SIValue lhs_value = AR_EXP_Evaluate(node->cond.left->exp.exp, NULL);
+		// If lhs is false, everything is false.
+		if(!lhs_value.longval) {
+			FilterTree_Free(node->cond.left);
+			FilterTree_Free(node->cond.right);
+			_FilterTree_In_Place_Set_Exp(node, lhs_value);
```

Comment:
it's either an in place update or work with a pointer to a pointer.
`_FilterTree_Compact_And(FT_FilterNode **node)`

Reply:
in place